tephra-hbase-compat-1.1-base/src/test/java/org/apache/tephra/hbase/txprune/InvalidListPruneTest.java

```diff
@@ -334,6 +337,45 @@ public Table get() throws IOException {
     }
   }
 
+  @Test(timeout = 60000L)
+  public void testClusterShutdown() throws Exception {
+    java.util.concurrent.ExecutorService executorService = Executors.newSingleThreadExecutor();
+    try {
+      // Create a new transaction snapshot
+      InMemoryTransactionStateCache.setTransactionSnapshot(
+        new TransactionSnapshot(100, 100, 100, ImmutableSet.of(50L),
+                                ImmutableSortedMap.<Long, TransactionManager.InProgressTx>of()));
+      // Run major compaction
```

Comment:
The comment can be changed to - `Try to create a situation where HBase shuts down when the async prune writer thread is trying to write to the prune table on a compaction. This will surface any deadlocks during HBase shutdown.`